articles/javascript-iterations-which-one-is-fast/index.md

```diff
@@ -0,0 +1,657 @@
+for (let loop = 0; loop < 5; loop++){
+console.log("hello world") //loop body
+}
+```
+
+### Do While
+Repeats a statement until the specified condition evaluates to false.
+```js
+do {
+  statement; //loop body
+}  while (condition);
+```
+The statement is checked/execute before the condition. If the condition is true, the statement loop will be executed again. In the do-while loop, the next statement executes before the condition rechecks the previously executed statement. If the condition is true, the statement is executed again until the condition evaluates to false.
+```js
+let i = 0;
+do {
+console.log("hello world");
+  i ++;
+} while (i < 5);
+```
+In the above example, `do` iterates before `while` then `i` reiterates until the condition executes to false. I.e. until the loop cycle is no longer less than five `(i < 5)`.
+
+### While Loop
+Loops as long as the condition evaluates to true. When the condition evaluates to false, the whole loops statement stop. While check a condition before executing the specified statement. If the statement executes, the loop retests the condition again. If the condition will be false at the first loop cycle, there is a chance the loop will not call a statement.
+```js
+let i = 0;
+while (i < 5) {
+  i++;
+  console.log(i);
+}
+```
+The above block will run until `i < 5` is not valid, and the loop will stop when the value of `i` is `5`. Otherwise, if the condition is always true, the loop will never end. Thus creating an infinite loop. I.e:
+```js
+let i = 0;
+while (true) {
+  i++;
+  console.log('Hello, world!');
+}
+```
+The above code block will execute forever, as the condition will never be false.
+
+### For In
+Iterates over named properties of an object key and get each property value inside the loop. A for-in loop has no condition expression and incremental value. The loop executes once per property. And tests are many times as the number of properties in an object.
+```js
+for(variable in object) {
+//statement
+}
+```
+The main advantage of for in over the other loops its capacity to loop through an object.
+```js
+const person = {
+name: "bunny",
+age:" 28",
+city: "LA",
+maritalstatus: true
+}
+for(let info in person)
+console.log(`${info}: ${person[info]}`)
+```
+For in will invoke the above loop as many times as the number of properties you add in object person. For in is a better solution to loop objects.
+
+### For Of
+Iterate through iterables such as `strings`, `arrays`, `maps`, and `set`. For of loops iterate through items in a collection such as an array. It is very similar to the for in except this time we are using the “of” keyword and passing it an iterable instead of an object. For of will not work with Objects because they are not iterable.
+```js
+for(variable of iterable) {
+    //statement
+}
+```
+For in also work with arrays, it uses the index of an array of each value. The following example sample the results difference of for in and for of to iterate over array elements.
```

Comment:
You could discuss all the uses of `for in` under the `#### For In` section as it would be clearer than reintroducing it at this point

Reply:
@LinusMuema , I included this section so that the difference between `for-of`  and `for-in`  can standout.